### DISCUSSION TITLE
Implementing Clerk Auth using Ignite (without demo code)

### DISCUSSION THREAD

**Discussion by niteshc132** [infinitered/ignite#2687] [Q&A]
Hey, im having trouble implementing Clerk using Ignite, I could putting the ClerkProvider in the wrong place maybe?

I tried putting the Clerk wrapper in the `app.tsx`

like this 
`    <ClerkProvider publishableKey={"pk_test_"}>

      <SafeAreaProvider initialMetrics={initialWindowMetrics}>

        <ErrorBoundary catchErrors={Config.catchErrors}>

          <GestureHandlerRootView style={$container}>

            <SignedIn>

              <AppNavigator
                linking={linking}
                initialState={initialNavigationState}
                onStateChange={onNavigationStateChange}
              />

            </SignedIn>

            <SignedOut>
       
              <SignInScreen />
       
                 </SignedOut>

          </GestureHandlerRootView>

        </ErrorBoundary>

      </SafeAreaProvider>
   
 </ClerkProvider>`
    
    however, trying to sign in results in this error
    
![image](https://github.com/infinitered/ignite/assets/25761319/87b8a92f-7e71-4e14-8774-b124fffcd7e7)

Thank you

**Comment by morganick** [comment]
@niteshc132 take a look at this recipe: https://ignitecookbook.com/docs/recipes/Authentication

Though it uses Supabase as the backend, you should be able to translate it for your needs to work with Clerk. The goal was to make it easy to swap out the backend so you could bring your own.
